sled-agent/src/instance_manager.rs

```diff
@@ -804,9 +805,13 @@ impl InstanceManagerRunner {
         for (id, instance) in self.jobs.iter() {
             // If we can read the filesystem pool, consider it. Otherwise, move
             // on, to prevent blocking the cleanup of other instances.
-            let Ok(Some(filesystem_pool)) =
-                instance.get_filesystem_zpool().await
-            else {
+            // TODO(eliza): clone each instance and spawn a task to handle it,
+            // so that a single misbehaving instance cannot block the instance
+            // manager's run loop...
```

Comment:
This might be a better comment on the follow-up PR, but I'm having trouble envisioning how this will work. Today this routine won't return until it's disposed of every VMM in `jobs`. Delegating to individual tasks will allow this loop to move on to consider VMM Y even if requests to VMM X aren't currently being serviced, but won't this routine still need to join all of the tasks at the end?

Not a blocker for this PR; just trying to dispel the "what am I missing?" feeling :)

Reply:
I was imagining that the instances that must be disposed of would be cloned into a task that would also own the request and it would just go and do the whole thing and then complete the request when it finishes, so that instances which don't need to be destroyed can go on handling requests to do other stuff whilst the disk expungement is being handled. This is in contrast to forcing requests to instances which know nothing about the expunged disk to wait until the expungement has finished. Does that make sense?

Reply:
It does, thanks!

The bit I was headed toward is that this ultimately means it will be possible to have an instance manager servicing two separate "use only these disks" commands at once, which is new behavior. That may be fine (I haven't thought about it) but I wanted to make sure I understood what was happening before actually raising it as something worth considering.